source/src/apps/public/analysis/SID_ERMS_Rescore.cc

```diff
@@ -0,0 +1,673 @@
+core::Real Rescore_models( const core::Real nscore_, const core::Real RMSE_, const core::Real max_, const core::Real min_ ){
+	//score structure
+	core::Real rescore, midpnt, e_exp;
+	midpnt = (max_ + min_ )/2;
+	e_exp = (50 * RMSE_) - (0.9 * 50 * midpnt);
+	rescore = 1000 * ( 1 - ( 1 /(0.1 + ( exp( e_exp ) ) ) ) );
+	if (rescore < 0) {
+		rescore = 0;
+	}
+
+	rescore += nscore_;
+	return rescore;
+}
+
+int
+main( int argc, char * argv [] )
+{
+
+	try{
+		using namespace basic::options;
+		using namespace basic::options::OptionKeys;
+		using namespace core::scoring;
+		using namespace utility;
+		using namespace core;
+		using namespace core::pose;
+
+	//inputting all options
+
+		//hold over local options for ERMS prediction
+		option.add( complex_type, "TSV file containing the information on complex type (nodes and edges)" ).def("comp.tsv");
+		option.add( ERMS, "TSV file containing the ERMS data or acceleration energies" ).def("ERMS.tsv");
+		option.add( RMSE, "Calculate RMSE?" ).def(false);
+		option.add( B_vals, "TSV file containing the B values (eV) for the interfaces" ).def("B.tsv");
+		option.add( steepness, "Steepness value (A) as input by the user" );
+		option.add( breakage_cutoff, "Breakage cutoff for simulation (eV)" ).def(0.0);
+
+		devel::init( argc, argv );
+
+	//checking and reading all inputs
+
+		core::Real A_(0.0025);
+
+		//breakage cutoff
+		core::Real breakage_cutoff_(0.0);
+		if ( option[ breakage_cutoff ].user() ) {
+			breakage_cutoff_ = option[ breakage_cutoff ]();
+			if ( breakage_cutoff_ < 0 ) {
+				std::stringstream err_msg;
+				err_msg << "Breakage cutoff must be greater than zero. Value of " << breakage_cutoff_ << " eV given." << std::endl;
+				utility_exit_with_message(err_msg.str());
+			}
+			A_ = 0.002; //default steepness when using breakage cutoff
+		}
+
+		//steepness parameter
+		if ( option[ steepness ].user() ) {
+			A_ = option[ steepness ]();
+			TR.Warning << "Custom steepness of " << A_ << " input. Not recommended." << std::endl;
+		}
+
+		//complex type (nodes and edges)
+		core::Size n_chains_;
+		utility::vector1<char> nodes_;
+		utility::vector1<utility::vector1<std::string>> edges_;
+		if ( option[ complex_type ].user() ) {
+			read_complex_type(n_chains_, nodes_, edges_);
+		} else {
+			std::stringstream err_msg;
+			err_msg << "Must input the complex type using the complex_type option." << std::endl;
+			utility_exit_with_message(err_msg.str());
+		}
+		TR << "Complex type read in" << std::endl;
+
+		//create score function object
+		core::scoring::ScoreFunctionOP sfxn = core::scoring::get_score_function();
+
+		std::ostringstream out;
+
+		out << "Structure" << "\t" << "RMSE" << "\t" << "Ref15_Score" << "\t" << "ERMS Rescore" << std::endl;
+
+		utility::vector1<core::Real> rmse_list;
+		utility::vector1<core::Real> score_list;
+
+		//pose specific input checks
+
+		utility::vector1<std::string> files;
+		utility::vector1<file::FileName> list;
+		if ( option[in::file::l].user() ) {
```

Comment:
I'd probably recommend using `streams_from_cmd_line()` from core/import_pose/pose_stream/util.hh -- this parses the command line and loads the poses for you.  (And is flexible with various inputs: -s/-l/-in:file:silent/etc.)

```
MetaPoseInputStream input = streams_from_cmd_line();
core::pose::Pose current_pose;
while ( input.has_another_pose() ) {
    input.fill_pose( current_pose, *rsd_set );

    std::cout << "Read pose for file: " << core::pose::tag_from_pose(current_pose) << std::endl;
}
```